app/serializers/base_bookmarkable.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+##
+# Anything that we want to be able to bookmark must be registered as a
+# bookmarkable type using Bookmark.register_bookmarkable(bookmarkable_klass),
+# where the bookmarkable_klass is a class that implements this BaseBookmarkable
+# interface. Some examples are TopicBookmarkable and PostBookmarkable.
+#
+# These methods are then called by the Bookmarkable class through a public
+# interface, used in places where we need to list, send reminders for,
+# or otherwise interact with bookmarks in a way that is unique to the
+# bookmarkable type.
+class BaseBookmarkable
+  attr_reader :model, :serializer, :preload_associations
+
+  def initialize(model, serializer)
+    @model = model
+    @serializer = serializer
+  end
+
+  def list_query(user, guardian)
+    throw NotImplementedError
```

Comment:
Just curious, why did we choose `throw` instead of `raise` here? I think if our intent is to raise an exception here, then `raise` should be used instead.

Reply:
Ah yeah will fix, no reason just a brain fart